builder/dockerfile/parser/parser.go

```diff
@@ -171,11 +171,20 @@ func Parse(rwc io.Reader, d *Directive) (*Node, error) {
 		}
 		startLine := currentLine
 
+		warning := false
```

Comment:
`warning` is a bit ambiguous, in this context it could mean many things.

How about `blankLineWithinInstruction` ?